Update DynamoDB docs (#1208)
spiceai/docs: website/docs/components/data-connectors/dynamodb.md

|----------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------|
| `dynamodb_aws_region`            | Required. The AWS region containing the DynamoDB table                                                                                                 |
| `dynamodb_aws_access_key_id`     | Optional. AWS access key ID for authentication. If not provided, credentials will be loaded from environment variables or IAM roles                    |
| `dynamodb_aws_secret_access_key` | Optional. AWS secret access key for authentication. If not provided, credentials will be loaded from environment variables or IAM roles                |
| `dynamodb_aws_session_token`     | Optional. AWS session token for authentication                                                                                                         |
| `unnest_depth`                   | Optional. Maximum nesting depth for unnesting embedded documents into a flattened structure. Higher values expand deeper nested fields.                |

> | `schema_infer_max_records`       | Optional. Number of documents to use to infer the schema. Defaults to 10                                                                               |

Reviewer: ```suggestion
| `schema_infer_max_records`       | Optional. The number of documents to use to infer the schema. Defaults to 10                                                                               |
```

> | `scan_segments`                  | Optional. Number of segments for `Scan` request. 'auto' by default, which will calcualte nujmber of segments based on number of the records in a table |

Reviewer: ```suggestion
| `scan_segments`                  | Optional. Number of segments for `Scan` request. 'auto' by default, which will calculate number of segments based on number of the records in a table |
```

:::warning[Limitations]

- The DynamoDB connector will scan the first 10 items to determine the schema of the table. This may miss columns that are not present in the first 10 items.

> - DynamoDB connector does not support Decimal types.

Reviewer: ```suggestion
- The DynamoDB connector does not support Decimal types.
```
